### ISSUE TITLE
allow config file in $XDG_CONFIG_HOME/felix/config.yaml on macOS

### ISSUE THREAD

**Issue by jaschenb** [kyoheiu/felix#210]
Hey!

I organise all my config files in the $XDG_CONFIG_HOME directory even on macOS and I wanted to ask if you would be okay to add the feature that on macOS felix also looks for a config file in $XDG_CONFIG_HOME/felix/config.yaml
I saw that you use the general dirs function to get the config path, but would it be okay, to add a check if there is a config file in $XDG_CONFIG_HOME/felix and then load that if it exists?

**Comment by kyoheiu** [comment]
I think it's possible, and yeah, that would be useful!
I'll work on this.

**Comment by kyoheiu** [comment]
Just to be clear: You mean `/Users/$USER/.config` by saying `$XDG_CONFIG_HOME`, right?

**Comment by jaschenb** [comment]
yes. For me that would be enough. However if you already change this I guess you can think about checking if someone set the `$XDG_CONFIG_HOME` environment variable and use that and if not then fall back to `/Users/$USER/.config`. I could imagine, that at some point someone might also want that added.

**Comment by kyoheiu** [comment]
Hm.. It seems I have to do some research about this.
Ultimately you can create a symlink to the config file path and store it wherever you want though.

**Comment by kyoheiu** [comment]
Implemented on `develop` branch (I don't test it on my own mac yet).

```
//On macOS, felix looks for 2 paths:
//First `$HOME/Library/Application Support/felix/config.yaml`,
//and if it fails,
//`$HOME/.config/felix/config.yaml`.
```

If you're interested, feel free to install from `develop` and tinker with it!
Note that some other behaviors are also changed: e.g. felix no more creates config file even if it's not found: It just launches with the default configuration, without creating new one.

**Comment by kyoheiu** [comment]
Merged & released in v2.3.0.

**Comment by jaschenb** [comment]
Sorry that I didn't responst, but I was away. Thanks for implementing this
